### PR TITLE
[Angular] Use viewChild signal

### PR DESCRIPTION
**Pull request by qmonmert**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/angular/templates/src/main/webapp/app/login/login.component.ts.ejs [hunk 1]
```diff
@@ -56,7 +55,7 @@ export default class LoginComponent implements OnInit, AfterViewInit {
   }
 
   ngAfterViewInit(): void {
-    this.username.nativeElement.focus();
+    this.username()?.nativeElement.focus();
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/login/login.component.ts.ejs:58]:
```suggestion
    this.username().nativeElement.focus();
```

File: generators/angular/templates/src/main/webapp/app/login/login.component.ts.ejs [hunk 2]
```diff
@@ -31,8 +31,7 @@ import { AccountService } from 'app/core/auth/account.service';
   templateUrl: './login.component.html',
 })
 export default class LoginComponent implements OnInit, AfterViewInit {
-  @ViewChild('username', { static: false })
-  username!: ElementRef;
+  username: Signal<ElementRef | undefined> = viewChild('username');
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/login/login.component.ts.ejs:34]:
```suggestion
  username = viewChild.required<ElementRef>('username');
```

File: generators/angular/templates/src/main/webapp/app/login/login.component.ts.ejs [hunk 3]
```diff
@@ -16,7 +16,7 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-import { Component, ViewChild, OnInit, AfterViewInit, ElementRef, inject, signal } from '@angular/core';
+import { Component, OnInit, AfterViewInit, ElementRef, inject, signal, viewChild, Signal } from '@angular/core';
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/login/login.component.ts.ejs:19]:
```suggestion
import { Component, OnInit, AfterViewInit, ElementRef, inject, signal, viewChild } from '@angular/core';
```

File: generators/angular/templates/src/main/webapp/app/login/login.component.spec.ts.ejs [hunk 4]
```diff
@@ -110,7 +110,7 @@ describe('LoginComponent', () => {
       const node = {
         focus: jest.fn(),
       };
-      comp.username = new ElementRef(node);
+      comp.username = signal<ElementRef | undefined>(new ElementRef(node));
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/login/login.component.spec.ts.ejs:113]:
```suggestion
      comp.username = signal<ElementRef>(new ElementRef(node));
```

**Review by mshima — COMMENTED** [review]
Can be simplified?

**Review by DanielFran — APPROVED** [review]

**Review by mshima — APPROVED** [review]

**Pull request merged**